pkg/stub/handler.go

```diff
@@ -2,96 +2,531 @@ package stub
+		}
 		if err != nil {
-			logrus.Errorf("failed to get image client : %v", err)
-			return err
+			s = nil
+		}
+		if s != nil {
+			logrus.Printf("updating dockerconfig secret %s in openshift namespace", v1alpha1.SamplesRegistryCredentials)
+			_, err = h.secretclientwrapper.Update("openshift", &secretToCreate)
+		} else {
+			logrus.Printf("creating dockerconfig secret %s in openshift namespace", v1alpha1.SamplesRegistryCredentials)
+			_, err = h.secretclientwrapper.Create("openshift", &secretToCreate)
+		}
+		if err != nil {
+			return h.processError(samplesResource, v1alpha1.SecretUpdateFailed, err, "failed to create/update registry dockerconfig secret in openshif namespace : %v")
+		}
+		newStatus = corev1.ConditionTrue
+		h.registrySecret = secret
+	}
+
+	err := h.GoodStatusUpdate(samplesResource, newStatus, v1alpha1.ImportCredentialsExist)
+	if err != nil {
+		return err
+	}
+
+	h.registrySecret = secret
+	h.samplesResource = samplesResource
+
+	return nil
+}
+
+func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
+	switch event.Object.(type) {
+	case *corev1.Secret:
+		dockercfgSecret, _ := event.Object.(*corev1.Secret)
+
+		if dockercfgSecret.Name != v1alpha1.SamplesRegistryCredentials {
+			return nil
 		}
-		logrus.Printf("image client %#v", imageclient)
-		templist, err := tempclient.Templates("openshift").List(metav1.ListOptions{})
-		logrus.Printf("template list %#v", templist)
-		/*err := sdk.Create(newbusyBoxPod(o))
-		if err != nil && !errors.IsAlreadyExists(err) {
-			logrus.Errorf("failed to create busybox pod : %v", err)
+
+		if h.samplesResource == nil {
+			if !event.Deleted {
+				h.registrySecret = dockercfgSecret
+			}
+			return nil
+		}
+
+		err := h.manageDockerCfgSecret(event.Deleted, h.samplesResource, dockercfgSecret)
+		if err != nil {
 			return err
-		}*/
+		}
+
+	case *v1alpha1.SamplesResource:
+		newStatus := corev1.ConditionTrue
+		var srcfg *v1alpha1.SamplesResource
+		if event.Deleted {
+			logrus.Println("SampleResource deleted")
+			h.samplesResource = nil
+			return h.CreateDefaultResourceIfNeeded()
+		} else {
+			srcfg, _ = event.Object.(*v1alpha1.SamplesResource)
+		}
+
+		// coordinate with timer's check on creating
+		// default resource ... looks at h.sampleResource,
+		// which is not set until this whole case is completed
+		h.mutex.Lock()
+		defer h.mutex.Unlock()
+
+		if h.samplesResource != nil &&
+			// in case a new, non-default resource is created
+			h.samplesResource.Name == srcfg.Name {
+			currVersion, _ := strconv.Atoi(h.samplesResource.ResourceVersion)
+			newVersion, _ := strconv.Atoi(srcfg.ResourceVersion)
+			if newVersion <= currVersion {
+				return nil
+			}
+		}
+
+		// if the secret event came in before the samples resource event,
+		// it could not be processed (though it would have been cached in
+		// the Handler struct);  process it now
```

Comment:
so why could it not be processed?  i think this logic is a holdover from before you switched to a hardcoded secret name,where you would have needed to see the samplesresource to determine the secret name.

Reply:
see rationale at https://github.com/openshift/cluster-samples-operator/pull/5#discussion_r217518521

a) create secret, then update conditions when we get samplesresource
b) create secret and update conditions at same time when we get samplesresource

I liked b) better